lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -737,13 +767,44 @@ def get_edgecolor(self):
         return self._maybe_depth_shade_and_sort_colors(super().get_edgecolor())
 
 
+def _get_data_scale(X, Y, Z):
+    """
+    Estimate the scale of the 3D data for use in depth shading
+
+    Parameters
+    ----------
+    X, Y, Z : masked arrays
+        The data to estimate the scale of.
+    """
+    # Account for empty datasets. Assume that X Y and Z have the same number
+    # of elements.
+    if not np.ma.count(X):
+        return 0
+
+    # Estimate the scale using the RSS of the ranges of the dimensions
+    # Note that we don't use np.ma.ptp() because we otherwise get a build
+    # warning about handing empty arrays.
```

Comment:
What do you mean by 'build warning'? This file isn't built...

Reply:
iirc, this was throwing a warning in the AppVeyor build CI

Reply:
It seems more like a bug in NumPy, but we seem to be pinning <2.1 on AppVeyor for reasons. Anyway, it's not a build warning.

The problem though is you get different results (the mask is ignored):
```
>>> X = np.ma.masked_array([2, 3 ,4], mask=[True, True, True])
>>> np.ma.ptp(X)
masked
>>> np.ptp(X)
masked_array(data=2,
             mask=False,
       fill_value=999999)
```

If that's acceptable, the comment should say that instead.

Reply:
Or even:
```
>>> X = masked_array([2, 3 ,4, .3, .4], mask=[True, True, True, False, False])
>>> np.ma.ptp(X)
np.float64(0.10000000000000003)
>>> np.ptp(X)
masked_array(data=3.7,
             mask=False,
       fill_value=1e+20)
```

Reply:
Thanks, those examples help a ton! We do want to take the mask into account, updated to use the ma method and we'll see if CI complains.

Reply:
It's looking good